File: src/InvertedIndex.java
import java.util.*;

public class InvertedIndex {
    public static HashMap<String, Set<String>> wordToDocumentMap = new HashMap<>();

    public static String[] wordSplitter(Map.Entry<String, String> file) {
        String content = file.getValue();
        return content.split("\\s");
    }

    public static void makeWordToDocumentMap(String[] words, String fileName) {
        for (String word : words) {
            if (wordToDocumentMap.containsKey(word)) {
                wordToDocumentMap.get(word).add(fileName);
            } else {
                Set<String> documentList = new HashSet<>();
                documentList.add(fileName);
                wordToDocumentMap.put(word, documentList);
            }
        }
    }
    public static void showFiles(HashMap<String, String> fileNameToContent) {
        for (Map.Entry<String, String> file : fileNameToContent.entrySet()) {
            String fileName = file.getKey();
            String[] words = wordSplitter(file);
            makeWordToDocumentMap(words, fileName);
        }
    }
}

File: phase5/SampleLibrary/InvertedIndex.cs
namespace SampleLibrary;

public class InvertedIndex : IInvertedIndex
{

    public Dictionary<string, List<string>> WordToDocumentMap { get; } = new ();

    public void makeWordToDocumentMap(string[] words, string fileName) 
    {
        foreach(var word in words) 
        {
            if (WordToDocumentMap.ContainsKey(word)) 
            {
                WordToDocumentMap[word].Add(fileName);
            } else 
            {
                WordToDocumentMap[word] = new List<string> {fileName};
            }
        }
    }
    public void createIndex(Dictionary<string, string> fileNameToContent)
    {
        foreach (var keyValuePair in fileNameToContent)
        {
            var fileName = keyValuePair.Key;
            var words = keyValuePair.Value.Split();
            makeWordToDocumentMap(words, fileName);
        }
    }

}